m2cgen/ast.py

```diff
@@ -164,6 +223,14 @@ def __str__(self):
         args = ",".join([str(e) for e in self.exprs])
         return "VectorVal([" + args + "])"
 
+    def __eq__(self, other):
+        return (type(other) is VectorVal and
+                self.output_size == other.output_size and
+                all(i == j for i, j in zip(self.exprs, other.exprs)))
+
+    def __hash__(self):
+        return hash(tuple(e for e in self.exprs))
```

Comment:
I might be missing some important detail here but why can't we just pass `self.exprs` into `tuple` directly? I.e.: `tuple(self.exprs)`

Reply:
Fixed, thanks!